org.opent2t.sample.hub.superpopular/com.wink.hub/js/thingTranslator.js

```diff
@@ -0,0 +1,187 @@
+/* jshint sub:true */
+// This code uses ES2015 syntax that requires at least Node.js v4.
+// For Node.js ES2015 support details, reference http://node.green/
+
+"use strict";
+var request = require('request-promise');
+
+/**
+* This translator class implements the "Hub" interface.
+*/
+class Translator {
+    constructor(accessToken) {
+        this._accessToken = accessToken;
+
+        this._baseUrl = "https://api.wink.com";
+        this._devicesPath = '/users/me/wink_devices';
+
+        this._name = "Wink Hub"; // TODO: Can be pulled from OpenT2T global constants. This information is not available, at least, on wink hub.
+    }
+
+    /**
+     * Get the list of devices discovered through the hub.
+     */
+    getDevicesAsync() {
+        return this._makeRequest(this._devicesPath, 'GET').then((data) => {
+            var devices = data.data;
+
+            var filteredDevices = [];
+            devices.forEach((winkDevice) => {
+                // get the opent2t schema and translator for the wink device
+                var opent2tInfo = this._getOpent2tInfo(winkDevice);
+
+                if ((winkDevice.model_name !== 'HUB')  &&  // Skip hub itself. (WINK specific check, of course!)
+                    opent2tInfo != undefined) // we support the device
+                {
+                    // we only need to return certain properties back
+                    var device = {};
+                    device.name = winkDevice.name;
+
+                    // set the specific device object id to be the id
+                    device.id = this._getDeviceId(winkDevice);;
+
+                    // set the opent2t info for the wink device
+                    device.openT2T = opent2tInfo;
+                    
+                    filteredDevices.push(device);
+                }
+            });
+            return filteredDevices;
+        });
+    }
+
+    /**
+     * Get the name of the hub.
+     */
+    getName() {
+        return this._name;
+    }
+
+    /**
+     * Gets device details (all fields), response formatted per http://docs.winkapiv2.apiary.io/
+     */
+    getDeviceDetailsAsync(deviceType, deviceId) {
+
+        // build request URI
+        var requestPath = '/' + deviceType + '/' + deviceId;
+
+        // Make the async request
+        return _makeRequest(requestPath, 'GET');
+    }
+
+    /**
+     * Puts device details (all fields) payload formatted per http://docs.winkapiv2.apiary.io/
+    */
+    putDeviceDetailsAsync(deviceType, deviceId, putPayload) {
+
+        // build request path and body
+        var requestPath = '/' + deviceType + '/' + deviceId;
+        var putPayloadString = JSON.stringify(putPayload);
+
+        // Make the async request
+        return _makeRequest(requestPath, 'PUT', putPayloadString);
+    }
+
+    /** 
+     * Given the hub specific device, returns the opent2t schema and translator
+    */
+    _getOpent2tInfo(winkDevice) {
+        if (winkDevice.thermostat_id != undefined) {
```

Comment:
>!= [](start = 37, length = 2)

The linter will catch this: use `!==` instead of `!=`. And yes, the difference is significant:

```
null != undefined   // false
null !== undefined  // true
```